### PR TITLE
[CS] Implement Some Stuff

### PR DESCRIPTION
**Pull request by sebmarkbage**
Actual implementation for CS.

### CODE REVIEW

File: packages/react-cs-renderer/src/ReactNativeCS.js [hunk 1]
```diff
@@ -227,4 +247,48 @@ injectInternals({
...
+      props.children,
+      state.root,
+      null,
+      null,
+    );
+    if (scheduledCallback) {
+      const callback = scheduledCallback;
+      scheduledCallback = null;
+      callback(infiniteDeadline);
+    }
```
Comment (sebmarkbage) [inline packages/react-cs-renderer/src/ReactNativeCS.js]: @acdlite This is basically a hack around the fact that we can't just call performWork on a per root basis manually.

File: packages/react-cs-renderer/src/ReactNativeCSTypes.js [hunk 2]
```diff
@@ -5,19 +5,26 @@
  * LICENSE file in the root directory of this source tree.
  *
  * @flow
+ * @providesModule ReactNativeCSTypes
```
[the remark this answers is not in the document]
Comment (gaearon) [reply packages/react-cs-renderer/src/ReactNativeCSTypes.js]: I guess we'll need to add it and `react-native-renderer/src/ReactNativeTypes.js` to [this list](https://github.com/facebook/react/blob/cc54b6f48a532bf2073e8f7d888c6a3fcf416b63/scripts/circleci/check_modules.sh#L6):

```
EXPECTED='scripts/rollup/header.js
packages/react-native-renderer/src/ReactNativeTypes.js
packages/react-cs-renderer/src/ReactNativeCSTypes.js'
```

File: packages/react-cs-renderer/src/ReactNativeCS.js [hunk 3]
```diff
@@ -227,4 +246,48 @@ injectInternals({
   rendererPackageName: 'react-cs-renderer',
 });
 
-module.exports = ReactNativeCSFiber;
+type ReactCSProps = {children: ReactNodeList};
+type ReactCSState = {
+  root: Object,
```
Comment (acdlite) [inline packages/react-cs-renderer/src/ReactNativeCS.js]: This is a FiberRoot, right? Can we use a an existential operator? And maybe a comment. Once we upgrade Flow, we should export an opaque type.

**Pull request merged**